WooCommerce/Classes/ViewModels/Order Details/Receipts/ReceiptEligibilityUseCase.swift

```diff
@@ -0,0 +1,49 @@
+import Yosemite
+import Experiments
+
+final class ReceiptEligibilityUseCase {
+    private let stores: StoresManager
+    private let featureFlagService: FeatureFlagService
+    
+    private var siteID: Int64 {
+        stores.sessionManager.defaultStoreID ?? 0
+    }
+    
+    init(stores: StoresManager = ServiceLocator.stores,
+         featureFlagService: FeatureFlagService = ServiceLocator.featureFlagService) {
+        self.stores = stores
+        self.featureFlagService = featureFlagService
+    }
+
+    func isEligibleForBackendReceipts(onCompletion: @escaping (Bool) -> Void) {
+        guard featureFlagService.isFeatureFlagEnabled(.backendReceipts) else {
+            onCompletion(false)
+            return
+        }
+        
+        let action = SystemStatusAction.fetchSystemPlugin(siteID: siteID, systemPluginName: Constants.wcPluginName) { wcPlugin in
+            // 1. WooCommerce must be installed and active
+            guard let wcPlugin = wcPlugin, wcPlugin.active else {
```

Comment:
I don't think we really need the `.active` check here, I mostly added it for return early consistency like we do with other plugin checks, since Woo could be inactive but the user being previously logged in the app.